File: 프로그래머스/lv1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

int solution(int n)
{
    int answer = 0;
    string a = "";
    while (n >= 3)
    {
        a.append(to_string(n % 3));
        n = n / 3;
    }
    if (n == 1)
        a.append(to_string(n));
    else if (n == 2)
        a.append(to_string(n));

    reverse(a.begin(), a.end());

    if (a[0] == '1')
        answer += 1;
    else if (a[0] == '2')
        answer += 2;
    
    
    for (int i = 1; i < a.length(); i++)
    {
        if (a[i] == '1')
            answer += 1 * pow(3, i);
        else if (a[i] == '2')
            answer += 2 * pow(3, i);
    }

    return answer;
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        
        String str = Integer.toString(n, 3);
        
        int p = 0;
        for (char c : str.toCharArray()) {
            if (c == '0') { p++; continue; }
            
            answer += (c - '0') * (int) Math.pow(3, p);
            p++;
        }
        
        
        return answer;
    }
}